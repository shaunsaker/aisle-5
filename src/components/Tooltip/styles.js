import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const styles = StyleSheet.create({
  container: {},
  triangleContainer: {
    position: 'absolute',
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export default styles;
