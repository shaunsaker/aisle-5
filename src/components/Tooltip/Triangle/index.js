import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';

import styles from './styles';

const propTypes = {
  height: PropTypes.number,
};

const defaultProps = {};

const Triangle = ({ height }) => {
  return (
    <View
      style={[
        styles.container,
        { borderLeftWidth: height, borderRightWidth: height, borderBottomWidth: height },
      ]}
    />
  );
};

Triangle.propTypes = propTypes;
Triangle.defaultProps = defaultProps;

export default Triangle;
