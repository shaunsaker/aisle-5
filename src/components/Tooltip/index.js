import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';

import styles from './styles';

import Touchable from '../Touchable';
import Snackbar from '../Snackbar';
import Triangle from './Triangle';

export default class Tooltip extends React.Component {
  constructor(props) {
    super(props);

    this.getTrianglePosition = this.getTrianglePosition.bind(this);

    this.triangleSize = 20;

    this.state = {};
  }

  static propTypes = {
    text: PropTypes.string,
    triangleOrientation: PropTypes.number,
    handlePress: PropTypes.func,
    testID: PropTypes.string,
  };

  static defaultProps = {};

  getTrianglePosition(orientation) {
    const position = {
      top: 0,
      right: 0,
      bottom: 0,
      left: 0,
    };
    const removePositions = {
      0: 'bottom',
      90: 'left',
      180: 'top',
      270: 'right',
    };
    const positionToRemove = removePositions[orientation];

    if (positionToRemove) {
      // NOTE: This is mutating
      delete position[positionToRemove];
    }

    if (orientation === 180) {
      // Special case for bottom oriented triangles
      position.bottom = this.triangleSize / 2;
    }

    return position;
  }

  render() {
    const { text, triangleOrientation, handlePress, testID } = this.props;
    const triangleRotateStyles = { transform: [{ rotate: `${triangleOrientation}deg` }] };
    const trianglePositionStyles = this.getTrianglePosition(triangleOrientation);

    return (
      <Touchable
        onPress={handlePress}
        style={[styles.container, { padding: this.triangleSize }]}
        testID={testID}
      >
        <Snackbar text={text} disabled />

        <View style={[styles.triangleContainer, triangleRotateStyles, trianglePositionStyles]}>
          <Triangle height={this.triangleSize} />
        </View>
      </Touchable>
    );
  }
}
