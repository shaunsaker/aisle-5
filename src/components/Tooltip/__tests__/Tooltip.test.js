import React from 'react';
import renderer from 'react-test-renderer';

import Tooltip from '..';

describe('Tooltip', () => {
  const spies = [];
  const horizontalOrientation = 90;

  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(<Tooltip />);

      expect(component).toMatchSnapshot();
    });
  });

  describe('methods', () => {
    it('should handle getTrianglePosition', () => {
      const component = renderer.create(<Tooltip />);
      const instance = component.getInstance();
      const trianglePosition = instance.getTrianglePosition(horizontalOrientation);

      expect(trianglePosition).toEqual({ bottom: 0, right: 0, top: 0 }); // FIXME: not declarative
    });
  });

  afterEach(() => {
    spies.forEach((spy) => {
      if (spy) {
        spy.mockClear();
      }
    });
  });
});
