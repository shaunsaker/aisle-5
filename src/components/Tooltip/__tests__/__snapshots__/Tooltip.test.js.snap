// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip renders renders with minimum required props 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "padding": 20,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#FEA116",
        "borderColor": null,
        "borderRadius": 5,
        "borderWidth": 0,
        "elevation": 12,
        "opacity": 1,
        "paddingHorizontal": 15,
        "paddingVertical": 10,
        "shadowColor": "#212121",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.33,
        "shadowRadius": 4,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Muli-SemiBold",
          "fontSize": 15,
          "textAlign": "center",
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "position": "absolute",
        },
        Object {
          "transform": Array [
            Object {
              "rotate": "undefineddeg",
            },
          ],
        },
        Object {
          "bottom": 0,
          "left": 0,
          "right": 0,
          "top": 0,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "borderBottomColor": "#FEA116",
            "borderLeftColor": "transparent",
            "borderRightColor": "transparent",
          },
          Object {
            "borderBottomWidth": 20,
            "borderLeftWidth": 20,
            "borderRightWidth": 20,
          },
        ]
      }
    />
  </View>
</View>
`;
